tests/test/util/test_queue.cpp

```diff
@@ -135,16 +142,98 @@ TEST_CASE("Test queue on non-copy-constructible object", "[util]")
     REQUIRE(fb.get() == 2);
 }
 
-TEST_CASE("Test queue timeout must be positive", "[util]")
+TEMPLATE_TEST_CASE("Test queue timeout must be positive",
+                   "[util]",
+                   IntQueue,
+                   FixedCapIntQueue)
 {
     int timeoutValueMs;
 
     SECTION("Zero timeout") { timeoutValueMs = 0; }
 
     SECTION("Negative timeout") { timeoutValueMs = -1; }
 
-    faabric::util::Queue<int> q;
+    TestType q;
     q.enqueue(10);
     REQUIRE_THROWS(q.dequeue(timeoutValueMs));
 }
+
+TEST_CASE("Test fixed capacity queue blocks if queue is full", "[util]")
+{
+    FixedCapIntQueue q(2);
+
+    q.enqueue(1);
+    q.enqueue(2);
+
+    // Enqueue with a short timeout so the operation fails quickly
+    REQUIRE_THROWS_AS(q.enqueue(100), QueueTimeoutException);
+}
+
+TEST_CASE("Test fixed capacity queue", "[util]")
+{
+    FixedCapIntQueue q(2);
+    auto latch = faabric::util::Latch::create(2);
+
+    std::thread consumerThread([&latch, &q] {
+        // Make sure we consume once to make one slot in the queue
+        latch->wait();
+        q.dequeue();
+    });
+
+    // Fill the queue
+    q.enqueue(1);
+    q.enqueue(2);
+    // Trigger the consumer thread to consume once
+    latch->wait();
+    // Check we can then enqueue a third time
+    q.enqueue(3);
+
+    if (consumerThread.joinable()) {
+        consumerThread.join();
+    }
```

Comment:
I think these tests could still do a better job of exercising the await-to-enqueue logic. 

Something like this is fairly certain to exercise it:

```
    FixedCapIntQueue q(2);
    int nMessages = 100;

    // Fast producer
    bool producerSuccess = false;
    std::thread producerThread([&q, nMessages, &producerSuccess] {
        for(int i = 0; i < nMessages; i++) {
            SLEEP_MS(1);
            q.enqueue(i);
        }
        
        producerSuccess = true;
    };
    
    // Slow consumer
    bool consumerSuccess = false;
    std::thread consumerThread([&latch, nMessages, &consumerSuccess] {
        for(int i = 0; i < nMessages; i++) {
            SLEEP_MS(100);
            int res = q.dequeue();            
            if(res != i) {
                return;
            }
        }
        
        consumerSuccess = true;
    });

    if (producerThread.joinable()) {
        producerThread.join();
    }

    if (consumerThread.joinable()) {
        consumerThread.join();
    }
    
    REQUIRE(producerSuccess);
    REQUIRE(consumerSuccess);
```

Reply:
:+1: I add this test but halve the sleeping time in the slower thread so the test doesn't take too long to run.